src/main/groovy/com/ullink/NuGetPush.groovy

```diff
@@ -11,19 +11,17 @@ class NuGetPush extends BaseNuGet {
 
     NuGetPush() {
         super('push')
-        conventionMapping.map "nupkgFile", { project.tasks.nugetPack.packageFile }
-        project.afterEvaluate {
-            if (nupkgFile) {
-                def tasks = project.tasks.withType(NuGetPack.class).findAll { it.packageFile == nupkgFile }
-                if (tasks.size() == 1)
-                    dependsOn tasks[0]
-            }
-        }
+    }
+
+    String getNugetPackOutputFile() {
```

Comment:
this doesn't seem consistent with the dependsOn logic change.

On one hand you removed the discovery of the matching nugetPack task for the nupkg nugetPush wants to push (you put instead opinionated dependency for default tasks, which I'm fine with), but on the other hand nupkgFile is now *only* defaulted if there's a single nugetPack task.